wasm/jit/compiler.go

```diff
@@ -8,86 +8,311 @@ import (
 	compileHostFunction(address wasm.FunctionAddress) error
+	// compileLabel notify compilers of the beginning of a label.
 	// Return true if the compiler decided to skip the entire label.
 	compileLabel(o *wazeroir.OperationLabel) (skipThisLabel bool)
-	// Followings are resinposible for compiling each wazeroir operation.
+	// compileUnreachable adds instructions to return to engine with jitCallStatusCodeUnreachable status.
+	// See wasm.OpcodeUnreachable.
 	compileUnreachable() error
+	// compileSwap adds instruction to swap the stack top value with the target in the Wasm value stack.
+	// The values are might be on registers or memory-stack at runtime, so compiler implementations
+	// emit instructions to swap values depending these locations.
 	compileSwap(o *wazeroir.OperationSwap) error
+	// compileGlobalGet adds instructions to read the value of the given index in the ModuleInstance.Globals
+	// and push the value onto the stack.
+	// See wasm.OpcodeGlobalGet.
 	compileGlobalGet(o *wazeroir.OperationGlobalGet) error
+	// compileGlobalSet adds instructions to set the top value on the stack to the given index in the ModuleInstance.Globals.
+	// See wasm.OpcodeGlobalSet.
 	compileGlobalSet(o *wazeroir.OperationGlobalSet) error
+	// compileBr adds instrctions to branch into the given label.
+	// See wasm.OpcodeBr.
 	compileBr(o *wazeroir.OperationBr) error
+	// compileBrIf adds instrctions to pops a value and branch into ".then" lable if the value equals 1.
+	// Otherwise, the code branches into ".else" label.
 	compileBrIf(o *wazeroir.OperationBrIf) error
+	// compileBrTable adds instructions to do br_table operation.
+	// A br_table operation has list of targets and default target, and
+	// this pops a value from the stack (called "index") and decide which branch we go into next
+	// based on the value.
+	//
+	// For example, assume we have operations like {default: L_DEFAULT, targets: [L0, L1, L2]}.
+	// If "index" >= len(defaults), then branch into the L_DEFAULT label.
+	// Othewise, we enter label of targets[index].
+	// See wasm.OpcodeBrTable.
 	compileBrTable(o *wazeroir.OperationBrTable) error
+	// compileCall adds instructions to call into a funcion of the given index.
+	// See wasm.OpcodeCall.
 	compileCall(o *wazeroir.OperationCall) error
+	// compileCallIndirect adds instructions to perform call_indirect operation.
+	// This consumes the one value from the top of stack (called "offset"),
+	// and make a function call against the function whose function address equals "table[offset]".
+	//
+	// Note: This is called indirect function call in the sense that the target function is indirectly
+	// determined by the current state (top value) of the stack.
+	// Therefore, two checks are performed at runtime before entering the target function:
+	// 1) If "offset" exceeds the length of table, "out of bounds table access" states (jitCallStatusCodeTableOutOfBounds) is returned.
```

Comment:
Maybe `ErrRuntimeInvalidTableAcces` is the right one now? > `jitCallStatusCodeTableOutOfBounds`

Reply:
clarified

https://github.com/tetratelabs/wazero/pull/209/commits/9f50bbe542cac5324fa7c3c686a2d9f68908306a

Reply:
Sorry, not it was `jitCallStatusCodeInvalidTableAccess` (edited)